libcaf/libcaf/repository.py

```diff
@@ -417,44 +416,109 @@ def log(self, tip: Ref | None = None) -> Generator[LogEntry, None, None]:
         except Exception as e:
             msg = f'Error loading commit {current_hash}'
             raise RepositoryError(msg) from e
-
+        
+    def _build_fsTree(self, path: Path, tree_hashes: dict[str, Tree]) -> str:
+        """
+        Builds a Tree structure from the filesystem in memory.
+        Populates 'tree_hashes' with hash -> Tree mappings.
+        Returns the hash of the root tree.
+
+        param path: The directory path to build the tree from.
+        param tree_hashes: A dictionary to populate with tree hashes and their corresponding Tree objects.
+        return: The hash of the root Tree object.
+        """
+        if not path.is_dir():
+             raise NotADirectoryError(f"{path} is not a directory")
+        
+        def _recurse(current_path: Path) -> str:
```

Comment:
Since this recursion can be quite deep and we don't really have a good bound for it, it's better to use a stack than implicit recursion